lineapy/graph_reader/artifact_collection.py

```diff
@@ -58,6 +59,7 @@ def __init__(
             raise ValueError(
                 f"Duplicated input parameters detected in {input_parameters}"
             )
+        self.db = get_context().executor.db
```

Comment:
would prefer if we initialized a new instance of the db. that way we dont have to get_context here which include a whole lot more.

Reply:
Will do!

Reply:
Discussed this further with @lionsardesai and we decided to pass DB directly to `ArtifactCollection` since the latter is supposed to focus on the logic of artifact graph refactor (instead of hosting a bunch more such as DB initiation). Also, `ArtifactCollection` is supposed to be used as internal structure, so DB can be passed down to it by a public API that uses `ArtifactCollection`.

Reply:
@mingjerli This alters the way `ArtifactCollection` gets initialized, and I updated relevant parts in codebase accordingly (including those you implemented). Let me know if you have questions/concerns about this.